go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -0,0 +1,300 @@
+			Type: "NodePort",
+			Selector: map[string]string{
+				"dask.org/cluster-name": taskName + "-cluster",
+				"dask.org/component":    "scheduler",
+			},
+			Ports: []v1.ServicePort{
+				{
+					Name:       "comm",
+					Protocol:   "TCP",
+					Port:       8786,
+					TargetPort: intstr.FromString("comm"),
+				},
+				{
+					Name:       "dashboard",
+					Protocol:   "TCP",
+					Port:       8787,
+					TargetPort: intstr.FromString("dashboard"),
+				},
+			},
+		},
+	}
+}
+
+func createJobSpec(image string, container *core.Container, jobContainer v1.Container, workerSpec daskAPI.WorkerSpec, schedulerSpec daskAPI.SchedulerSpec) daskAPI.DaskJobSpec {
+
+	return daskAPI.DaskJobSpec{
+		Job: daskAPI.JobSpec{
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					jobContainer,
+				},
+			},
+		},
+		Cluster: daskAPI.JobClusterSpec{
+			Spec: daskAPI.DaskClusterSpec{
+				Worker:    workerSpec,
+				Scheduler: schedulerSpec,
+			},
+		},
+	}
+}
+
+func (p daskResourceHandler) GetTaskPhase(ctx context.Context, pluginContext k8s.PluginContext, r client.Object) (pluginsCore.PhaseInfo, error) {
+	logPlugin, err := logs.InitializeLogPlugins(logs.GetLogConfig())
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+
+	return p.GetTaskPhaseWithLogs(ctx, pluginContext, r, logPlugin, " (User)")
+}
+
+func (daskResourceHandler) GetTaskPhaseWithLogs(ctx context.Context, pluginContext k8s.PluginContext, r client.Object, logPlugin tasklog.Plugin, logSuffix string) (pluginsCore.PhaseInfo, error) {
+	job := r.(*daskAPI.DaskJob)
+
+	pod, err := getJobPodFromJobResource(job, ctx)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+	
+	// Logic copied from the pod plugin
+	transitionOccurredAt := flytek8s.GetLastTransitionOccurredAt(pod).Time
+	info := pluginsCore.TaskInfo{
+		OccurredAt: &transitionOccurredAt,
+	}
+
+	if pod.Status.Phase != v1.PodPending && pod.Status.Phase != v1.PodUnknown {
+		taskLogs, err := logs.GetLogsForContainerInPod(ctx, logPlugin, pod, 0, logSuffix)
+		if err != nil {
+			return pluginsCore.PhaseInfoUndefined, err
+		}
+		info.Logs = taskLogs
+	}
+
+	switch pod.Status.Phase {
+	case v1.PodSucceeded:
+		return flytek8s.DemystifySuccess(pod.Status, info)
+	case v1.PodFailed:
+		return flytek8s.DemystifyFailure(pod.Status, info)
+	case v1.PodPending:
+		return flytek8s.DemystifyPending(pod.Status)
+	case v1.PodReasonUnschedulable:
+		return pluginsCore.PhaseInfoQueued(transitionOccurredAt, pluginsCore.DefaultPhaseVersion, "pod unschedulable"), nil
+	case v1.PodUnknown:
+		return pluginsCore.PhaseInfoUndefined, nil
+	}
+
+	if len(info.Logs) > 0 {
+		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion+1, &info), nil
+	}
+	return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &info), nil
+}
+
+
+func getClientset() (*kubernetes.Clientset, error) {
+	// FIXME: Handle auth
+	config, err := clientcmd.BuildConfigFromFlags("", "/Users/bstadlbauer/.kube/config")
```

Comment:
Note that this is only for testing, if we keep it, it would have a similar logic to the one in propeller (also supporting in-cluster auth, etc.)